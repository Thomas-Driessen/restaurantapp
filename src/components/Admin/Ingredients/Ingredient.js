import React from "react";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import IngredientEdit from "./IngredientEdit";
import IngredientDelete from "./IngredientDelete";

const Ingredient = (props) => {

    return (
        <div>
            {props.ingredient ? (
                <div>
                    <Card>
                        <CardContent>
                            <Typography gutterBottom variant="inherit" component="h2">
                                {props.ingredient.ingredientTitle}{" "}
                                <span style={{ float: "right" }}>
                                    <span style={{ color: "green" }}> {props.ingredient.price}â‚¬ </span> / 100g
                                </span>
                            </Typography>
                            <Typography component="h6">
                                Quantity: {props.ingredient.ingredientQuantity}
                            </Typography>
                        </CardContent>
                        <CardActions>
                            <Grid
                                container
                                alignItems="flex-start"
                                justify="flex-end"
                                direction="row"
                            >
                                <div style={{ display: "flex", alignItems: "right" }}>
                                    <IngredientEdit
                                        ingredient={props.ingredient}
                                    />
                                    <IngredientDelete
                                        ingredient={props.ingredient}
                                    />
                                </div>
                            </Grid>
                        </CardActions>
                    </Card>
                </div>
            ) : null}
        </div>
    );
};
export default Ingredient;
