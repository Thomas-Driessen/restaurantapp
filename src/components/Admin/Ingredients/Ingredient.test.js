import React from 'react';
import { render } from '@testing-library/react';
import Ingredient from './Ingredient';
import renderer from 'react-test-renderer';


const ingredient = {
  ingredientId: 1,
  ingredientTitle: "Salt",
  ingredientQuantity: 400,
  price: 2.5
};

it('Ingredient renders correctly', () => {
  const tree = renderer
    .create(<Ingredient ingredient={ingredient} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Ingredient', () => {
  const { getByText } = render(<Ingredient ingredient={ingredient} />);
  const linkElement = getByText(/Salt/i);
  expect(linkElement).toBeInTheDocument();
});

afterAll(() => {
  global.gc && global.gc()
})