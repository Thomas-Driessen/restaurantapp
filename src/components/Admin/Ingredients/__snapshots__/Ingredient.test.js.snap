// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredient renders correctly 1`] = `
<div>
  <div>
    <div
      className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-gutterBottom"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Salt
           
          <span
            style={
              Object {
                "color": "black",
                "float": "right",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
               
              2.5
              â‚¬ 
            </span>
             / 100
          </span>
        </h2>
        <h6
          className="MuiTypography-root MuiTypography-body1"
        >
          Milliliters
           in inventory: 
          400
        </h6>
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-flex-end"
        >
          <div
            style={
              Object {
                "alignItems": "right",
                "display": "flex",
              }
            }
          >
            <div>
              <button
                aria-label="edit"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div>
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
